kickstarter/android-oss
# Hadia/refactor replies adapter

# 📲 What

Refactor replies adapter to re-use on section and change the UI for view more and pagination error 

# 🤔 Why

we have crash for the section use
https://console.firebase.google.com/u/2/project/android-internal-debug/crashlytics/app/android:com.kickstarter.kickstarter.internal.debug/issues/326dd98632ecc319079610bd4ec5909b?time=last-hour&sessionEventKey=60EC4702003B000152CFB4810112C742_1562481805897445917

# 🛠 How
https://www.loom.com/share/cc4014ef9fbd4eefbee78e9f815f0e58
 re-use  section with status to show error
![image](https://user-images.githubusercontent.com/1075310/125300587-7a01c500-e32a-11eb-9cde-229924ca0ff2.png)
Create 3 Different adapters for each cell and use ConcatAdapter |
Rename class and refactor test 

# 👀 See

https://user-images.githubusercontent.com/1075310/125298325-52a9f880-e328-11eb-97f6-c30905ef8430.mp4

add the first reply


https://user-images.githubusercontent.com/1075310/125414003-4add2cf6-2794-4f32-a0e6-5680e3f07430.mp4



# 📋 QA

1. Open-backed Project add the first reply for comment
2. Open-backed Project  **"Elementallis - A Zelda-like with Elemental magic gameplay"** with a comment has pagination replies  
3. view more replies
4. go offline to show error then online to see it working and load more
5. add a new comment to check scrolling behavior 

# Story 📖

https://kickstarter.atlassian.net/browse/NT-2100
